### ISSUE TITLE
Memory unit for enhancing memorization in autoencoders

### ISSUE THREAD

**Issue by DanielWicz** [tensorflow/addons#2755]
**Describe the feature and the current behavior/state.**

The feature allows for inclusion of a new layer type called Memory Unit. This layer allows for efficient memorization of the data, that were used to train neural network. 
At the given moment, this unit is used in the MemAE (memory-augmented autoencoder). Nevertheless, if it would be more popularized, it could be used in other problems, the same as attention module got widespread after a while. 

**Relevant information**
- Are you willing to contribute it (yes/no): No
*If you wish to contribute, then read the requirements for new contributions in [`CONTRIBUTING.md`](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#requirements-for-new-contributions-to-the-repository)*
- Are you willing to maintain it going forward? (yes/no): No
- Is there a relevant academic paper? (if so, where): https://arxiv.org/abs/1904.02639
- Does the relavent academic paper exceed 50 citations? (yes/no):  yes (550 citations)
- Is there already an implementation in another framework? (if so, where):  https://github.com/donggong1/memae-anomaly-detection/blob/master/models/memory_module.py (class called MemoryUnit)
- Was it part of tf.contrib? (if so, where): No

**Which API type would this fall under (layer, metric, optimizer, etc.)**
layer
**Who will benefit with this feature?**
People who are using autoencoders for anomaly detection and enhance discovery in the field of data memorization in neural network

**Any other info.**

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
